File: Python/Linear Search.py
def linear_search(arr, target, steps):
    """
    Perform a linear search for the target value in the given array.
    
    :param arr: List of elements to search through.
    :param target: The element to search for.
    :param steps: A list containing a single integer that tracks the number of steps.
    :return: The index of the target element if found, otherwise -1.
    """
    for index, element in enumerate(arr):
        steps[0] += 1
        if element == target:
            return index
    return -1

def main():
    # Example list input
    my_list = list(map(int, input("Enter list elements separated by spaces: ").split()))
    
    # Target value input
    target_value = int(input("Enter target value: "))
    
    # Time complexity O(N)
    steps = [0]  # Using a list to track the number of steps
    
    # Perform linear search
    index = linear_search(my_list, target_value, steps)
    
    # Display the result
    if index != -1:
        print(f"Element {target_value} found at index: {index}")
    else:
        print(f"Element {target_value} not found in the list.")
    
    # # Time Complexity is O(n)
    print(f"Total steps: {steps[0]}")

if __name__ == "__main__":
    main()

File: Java/Linear_Search.java
public class Linear_Search {
    public static void main(String[] args) {
        
        // linear search = iterative through collection one element at a time .. time complexity is O(N)
        int [] array = {5, 1, 9, 2, 6, 4, 3, 0};

        int index = linearSearch(array, 1);

        if (index != -1){
            System.out.println("Element found at index: " + index);
        }
        else {
            System.out.println("Element not found at index");

        }
    }

    private static int linearSearch(int[] array, int value){

        for (int i = 0; i < array.length; i++){
            if (array[i] == value) {
                return i;
            }
        }
        return -1;
    }
}